packages/compile-common/src/compilations.ts

```diff
@@ -0,0 +1,9 @@
+import type { CompilerResult, WorkflowCompileResult } from "./types";
+
+export function promoteCompileResult(
+  result: CompilerResult
+): WorkflowCompileResult {
+  const { compilations } = result;
+  const contracts = compilations.flatMap(compilation => compilation.contracts);
```

Comment:
I knew I shouldn't have mentioned category theory earlier 😅

Reply:
I've been waiting to finally be able to use `flat()` and `flatMap()` for quite a while now :)